src/mixins/triangle.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+
+/**
+ * CSS to represent triangle with any pointing direction.
+ *
+ * @example
+ * // Styles as object usage
+ *
+ * const styles = {
+ *   ...triangle({ pointing: 'right', width: '100px', height: '100px'})
+ * }
+ * const styles = {
+ *   ...triangle({ pointing: 'right', width: '100px', height: '100px'})
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   ${triangle({ pointing: 'right', width: '100px', height: '100px'})}
+ *
+ *
+ * // CSS as JS Output
+ *
+ * div: {
+ *  'border-color': 'transparent',
+ *  'border-left-color': 'red !important',
+ *  'border-style': 'solid',
+ *  'border-width': '50px 0 50px 100px',
+ *  'height': '0',
+ *  'width': '0',
+ * }
+ */
+
+type PointingDirection = 'top' | 'right' | 'bottom' | 'left'
+
+type TriangleArgs = {
+  pointingDirection: PointingDirection,
+  color: string,
+  height: number,
+  width: number,
+}
+
+type TriangleSizes = {
+  height: number,
+  width: number,
+}
+
+const getLeftPointingTriangle = ({ height, width } : TriangleSizes) => (
+  `${height}px ${width}px ${height}px 0`
+)
+
+const getRightPointingTriangle = ({ height, width } : TriangleSizes) => (
+  `${height}px 0 ${height}px ${width}px`
+)
+
+const getTopPointingTriangle = ({ height, width } : TriangleSizes) => (
+  `0 ${width}px ${height}px ${width}px`
+)
+
+const getBottomPointingTriangle = ({ height, width } : TriangleSizes) => (
+  `${height}px ${width}px 0 ${width}px`
+)
```

Comment:
What about refactoring this to a single function? Something like

```JS
const getBorder = ({ height, width, direction }): TriangleSizes) => {
  switch (direction) {
    case 'left': return `${height}px ${width}px ${height}px 0`
    case 'right': return `${height}px 0 ${height}px ${width}px`
    case 'top': return `0 ${width}px ${height}px ${width}px`
    case 'bottom': return `${height}px ${width}px 0 ${width}px`
  }
}
```

This would allow us to get rid of a bunch of `if`s in the main function. 😊